cmd/run.go

```diff
@@ -167,15 +174,15 @@ func run(cmd *cobra.Command, args []string) error {
 		return fmt.Errorf("no command arguments passed, please specify or use --no-command option")
 	}
 
-	block, err := intoto.InTotoRun(stepName, runDir, materialsPaths, productsPaths, args, key, []string{"sha256"}, exclude, lStripPaths, lineNormalization, followSymlinkDirs)
+	metadata, err := intoto.InTotoRun(stepName, runDir, materialsPaths, productsPaths, args, key, []string{"sha256"}, exclude, lStripPaths, lineNormalization, followSymlinkDirs, useDSSE)
 	if err != nil {
 		return fmt.Errorf("failed to create link metadata: %w", err)
 	}
 
-	linkName := fmt.Sprintf(intoto.LinkNameFormat, block.Signed.(intoto.Link).Name, key.KeyID)
+	linkName := fmt.Sprintf(intoto.LinkNameFormat, metadata.GetPayload().(intoto.Link).Name, key.KeyID)
```

Comment:
`metadata.GetPayload().(intoto.Link)` add check for this type assertion to not cause a panic

Reply:
InTotoRun can only return a link, should we add the check nonetheless?

Reply:
I am ok with skipping the check if it will always be one type. Just in case it could return other types in the future, we would have to add the check.

Reply:
Agreed but we'd have to handle other things at the time anyway including where to get the `Name` etc. I'd prefer to handle this separately when the time comes. I think we may have a world where `InTotoRun` generates other supported predicates.